core/net/rpl/rpl-dag.c

```diff
@@ -1023,8 +1024,9 @@ rpl_add_dag(uip_ipaddr_t *from, rpl_dio_t *dio)
   PRINTF("\n");
 
   ANNOTATE("#A join=%u\n", dag->dag_id.u8[sizeof(dag->dag_id) - 1]);
-
-  rpl_process_parent_event(instance, p);
+  if(dag->rank != ROOT_RANK(instance)) {
+    rpl_process_parent_event(instance, p);
+  }
```

Comment:
rpl_process_parent_event() update the rank of the node, assuming it's a RPL router and not a Root. All the other calls to rpl_process_parent_event() are either protected the same way (rpl_process_dio() or global_repair()) or is never called (rpl_recalculate_ranks())

I will add a comment